dataprofiler/data_readers/json_data.py

```diff
@@ -244,15 +244,20 @@ def _load_data_from_file(self, input_file_path):
         :type input_file_path: str
         :return:
         """
-        self._file_encoding = data_utils.detect_file_encoding(input_file_path)
-        with open(input_file_path, encoding=self.file_encoding) as input_file:
+        self._file_encoding = None
+        if not isinstance(input_file_path, StringIO):
+            self._file_encoding = \
+                data_utils.detect_file_encoding(input_file_path)
+                
+        with FileOrBufferHandler(input_file_path, 'r', encoding=self.file_encoding) \
+            as input_file:
```

Comment:
I would suggest maybe breaking it up as:
```python
with FileOrBufferHandler(input_file_path, 'r', 
                                           encoding=self.file_encoding) as input_file:
```
If that fits.

Reply:
Otherwise, for reference when indenting, we would want the indented line over two indents rather than one in this case so that it isn't inline with the code below which may cause confusion.